test/gtest/uct/ib/test_cq_moderation.cc

```diff
@@ -34,31 +34,29 @@ class test_uct_cq_moderation : public uct_test {
             set_config("RC_FC_ENABLE=n");
         }
 
-        m_sender = uct_test::create_entity(0, NULL, NULL, NULL, NULL, NULL,
-                                           send_async_event_handler, this);
-        m_entities.push_back(m_sender);
-
-        m_receiver = uct_test::create_entity(0, NULL, NULL, NULL, NULL, NULL,
-                                             recv_async_event_handler, this);
-        m_entities.push_back(m_receiver);
+        uct_test::create_entity(0, NULL, NULL, NULL, NULL, NULL,
+                                send_async_event_handler, this);
+        uct_test::create_entity(0, NULL, NULL, NULL, NULL, NULL,
+                                recv_async_event_handler, this);
 
         check_skip_test();
 
-        m_send_async_event_ctx.wait_for_event(*m_sender, 0);
-        m_recv_async_event_ctx.wait_for_event(*m_receiver, 0);
+        m_send_async_event_ctx.wait_for_event(sender(), 0);
```

Comment:
is this change critical? seems using sender()/receiver() functions instead of member vars is done in multiple tests. can we split it to PRs by tests?

Reply:
> is this change critical?

refactoring is never critical

> seems using sender()/receiver() functions instead of member vars is done in multiple tests. can we split it to PRs by tests?

Technically yes, but there is a chance to lose some changes. From my perspective the pach is huge but simple and strade forward.